### PR TITLE
bear_updates.md: Improve description

### PR DESCRIPTION
**Pull request by NiklasMM**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

I hope it's not too late! Here's my take on improving the description for "Improve Bear Quality"

### CODE REVIEW

**Comment by adhikasp** [comment]
Looks great!

File: data/projects/bear_updates.md [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
+
+##### CODING PHASE 1
 
- * For all issues, discusions about how to solve them properly and sustainably
- have to be resolved.
- * Core changes and guidelines for enforcing higher bear quality are documented
- in [the coala repository](https://api.coala.io).
+ * Ideas for how to solve each of the indetified issues are drafted.
+ * Changes to the [linter bear documentation](http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html) to improve quality of future bears are proposed.
+ * A list of bears which suffer from any of the identified issues is composed.
```
Comment (sils) [inline data/projects/bear_updates.md]: none of those milestones are coding, they're all conceptual. We should really have a third of the project done *and merged* at this stage. Never pass a student without merged work ideally even for the community bonding (cEP drafting, docs stuff)

**Comment by sils** [comment]
the milestones need some work, rest looks good to me

File: data/projects/bear_updates.md [hunk 2]
```diff
@@ -1,26 +1,42 @@
...
+They do work, but not necessarily on all platforms or for all usecases. They are documented, but not
+necessarily understandable. They are tested, but lack corner cases or
 exploratory tests. [See existing issues](https://github.com/coala/coala-bears/issues?q=is%3Aopen+is%3Aissue+label%3Aarea%2Flintbears)
 
-Task of this project is to solve those issues for as many bears as possible and
-derive strategies to avoid those issues in the future.
+Students working on this project familiarize themselves in-depth with the working of the `linter`
+decorator and the existing linter bears. They identify systematic issues with the way linter bears are
+developed and maintained and work to fix those issues and improve and amend [existing documentation](http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html)
+to avoid them in the future.
```
Comment (sims1253) [inline data/projects/bear_updates.md]: This could probably have more punch to it. What would be the great outcome of this project? Probably something like a wider range of nice working bears and thus nobody has to encounter non working bears when using random bears we don't usually use

File: data/projects/bear_updates.md (hunk 1 above)
Comment (NiklasMM) [inline data/projects/bear_updates.md]: Do you have concrete suggestions? I don't want to write "some work should be merged". Also update of the docs is merged work.

Comment (sims1253) [inline data/projects/bear_updates.md]: this could be quite a lot of work considering the amount of bears we have.

File: data/projects/bear_updates.md [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
+ * Ideas for how to solve each of the indetified issues are drafted.
+ * Changes to the [linter bear documentation](http://api.coala.io/en/latest/Developers/Writing_Linter_Bears.html) to improve quality of future bears are proposed.
+ * A list of bears which suffer from any of the identified issues is composed.
 
-##### GSoC 2017 Midterm
+##### CODING PHASE 2
 
- * A given set of issues defined in the proposal are resolved.
+ * Working concepts of systematic fixes exists and can be demonstrated (e.g. a more convenient way to test, an automatic check for tool option support).
+ * A number of bears identified in the previous step have been improved to fit the new standards.
```
Comment (sims1253) [inline data/projects/bear_updates.md]: Make all milestones concrete and testable. So eg this could be be "at least one bear is fixed"

File: data/projects/bear_updates.md [hunk 4]
```diff
@@ -1,26 +1,42 @@
...
 
-##### GSoC 2017 Final
+##### CODING PHASE 3
 
- * A given set of issues defined in the proposal are resolved.
+* Any systematic fixes are finished and available for other bear developers to use.
+* If applicable a system is in place to automatically detect issues in newly created linter bears.
+* The bears from the list identified in coding phase 1 have all been fixed and improved to fit the new standards.
```
Comment (sims1253) [inline data/projects/bear_updates.md]: I don't think this is doable in the time frame considering the other milestones this has.

File: data/projects/bear_updates.md [hunk 5]
```diff
@@ -1,26 +1,42 @@
...
- * A given set of issues defined in the proposal are resolved.
+ * Working concepts of systematic fixes exists and can be demonstrated (e.g. a more convenient way to test, an automatic check for tool option support).
+ * A number of bears identified in the previous step have been improved to fit the new standards.
 
-##### GSoC 2017 Final
+##### CODING PHASE 3
 
- * A given set of issues defined in the proposal are resolved.
+* Any systematic fixes are finished and available for other bear developers to use.
+* If applicable a system is in place to automatically detect issues in newly created linter bears.
```
Comment (sims1253) [inline data/projects/bear_updates.md]: Maybe this could be part of phase 1 as a "a system to identify systematic problems is proposed" and then in phase 3 it is implemented?

File: data/projects/bear_updates.md (hunk 1 above)
Comment (NiklasMM) [inline data/projects/bear_updates.md]: Suggestions? Should we ask for a specific number?

Comment (sims1253) [inline data/projects/bear_updates.md]: not really
but milestones are not super important at this time

**Comment by sims1253** [comment]
ack e899f6f8618db6bd7a774db44dcd94e5c5b382f4

**Comment by NiklasMM** [comment]
rebased

**Comment by sils** [comment]
reack https://github.com/coala/projects/pull/198/commits/2681a15d97333c0d2976a500a3161b721a857835
